esm-loader.mjs

```diff
@@ -51,6 +58,9 @@ export async function resolve(specifier, context, nextResolve) {
       // Stripping the prefix is necessary because the code downstream gets this url without it
       instrumentationDefinitionCopy.moduleName = resolvedModule.url.replace('file://', '')
 
+      // Added to keep our Supportability metrics from exploding/including customer info via full filepath
+      instrumentationDefinitionCopy.friendlyModuleName = specifier
```

Comment:
I'm wondering if we should log a ticket to do the equivalent in the esm based versioned tests: [hapi onRequire versioned test assertions](https://github.com/newrelic/node-newrelic/commit/d56e4b004#diff-b8b56d39535a75db8ac563683afb072c0c5a4e030f6f9c402ce0a770db86fe91R41)

Reply:
I'm not sure if this is 100% parity with the cjs resolver stuff.  You could always do stuff like `require('node_modules/express')` but not sure if we just got `express` or the full path to what you required.  I know in esm land this is way more permissive. Have you explored what specifier is if you import modules with varieties in paths like relative, urls, etc?

Reply:
Did a quick test using 9.0.3, even in cjs, if you do something like have `express` installed in a parent folder and reference it with a relative path, like `const express = require('../../node_modules/express')`, you'll get that precise string `../../node_modules/express'`, which won't match the key `express` in `shimmer.registeredInstrumentations`, so `express` won't get instrumented.

Reply:
yea that's what i figured. we have a bigger issue with how we may instrumentation to require/import statements which we need to address in the future.  I'll put it on my list to capture as a ticket.

Reply:
Thinking more about this variable name.  I'm leaning towards just keeping it specifier. We're not used to it yet but it's the name that's referenced in all the ESM docs